subprojects/docs/src/docs/userguide/customTasks.adoc

```diff
@@ -250,6 +250,108 @@ Using Gradle's `IncrementalTaskInputs` is not the only way to create tasks that
 
 However, if the state files are non-relocatable, then they can't be shared via the build cache. Indeed, when the task is loaded from cache, any such state files must be cleaned up to prevent stale state to confuse the tool during the next execution. Gradle can ensure such stale files are removed if they are declared via `task.localState.register()` or a property is marked with the `@LocalState` annotation.
 
+[[sec:declaring_and_using_command_line_options]]
+=== Declaring and Using Command Line Options
+
+[NOTE]
+====
+The API for exposing command line options is an <<feature_lifecycle,incubating>> feature.
+====
+
+A custom task implementation exposes properties to make the runtime behavior configurable. Sometimes a user wants to declare the value of an exposed task property on the command line instead of the build script. That's particular helpful if the property value changes more frequently. The task API supports a mechanism for marking a property to automatically generate a corresponding command line parameter with a specific name at runtime.
+
+[[sec:declaring_task_option]]
+==== Declaring an option
+
+Exposing a new command line parameter for a task property is straightforward. You just have to annotate the corresponding setter method of a property with api:org.gradle.api.tasks.options.Option[]. An option requires a mandatory identifier. Additionally, you can provide an optional description. A task can expose as many command line options as properties available in the class.
+
+Let's have a look at an example to illustrate the functionality. The custom task `UrlVerify` verifies whether a given URL can be resolved by making a HTTP call and checking the response code. The URL to be verified is configurable through the property `url`. The setter method for the property is annotated with api:org.gradle.api.tasks.options.Option[].
+
+++++
+<sample id="stringBasedOption" dir="userguide/tasks/commandLineOption/stringOption/buildSrc/src/main/java" title="Declaring a command line option">
+    <sourcefile file="UrlVerify.java" snippet="custom-task-implementation"/>
+</sample>
+++++
+
+All options declared for a task can be <sec:listing_task_options,rendered as console output>> by running the `help` task.
+
+[[sec:using_task_option_command_line]]
+==== Using an option on the command line
+
+Using an option on the command line has to adhere to the following rules:
+
+- The option uses a double-dash as prefix e.g. `--url`. A single dash does not qualify as valid syntax for a task option.
+- The option parameter follows directly after the task declaration e.g. `verifyUrl --url=http://www.google.com/`.
```

Comment:
I weakly prefer the term "argument" to "parameter" here, but your call on this one.

Reply:
done